internal/cmd/plugin/logs/pretty/pretty.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pretty
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/cloudnative-pg/machinery/pkg/stringset"
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin"
+)
+
+type prettyCmd struct {
+	loggers   *stringset.Data
+	pods      *stringset.Data
+	groupSize int
+	verbosity int
+	minLevel  LogLevel
+}
+
+// NewCmd creates a new `kubectl cnpg logs pretty` command
+func NewCmd() *cobra.Command {
+	var loggers, pods []string
+	var groupSize, verbosity int
+	bf := prettyCmd{}
+
+	cmd := &cobra.Command{
+		Use:   "pretty",
+		Short: "Prettify CNPG logs",
+		Long:  "Reads CNPG logs from standard input and pretty-prints them for human consumption",
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return plugin.CompleteClusters(cmd.Context(), args, toComplete), cobra.ShellCompDirectiveNoFileComp
+		},
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			bf.loggers = stringset.From(loggers)
+			bf.pods = stringset.From(pods)
+			bf.groupSize = groupSize
+			bf.verbosity = verbosity
+
+			recordChannel := make(chan logRecord)
+			recordGroupsChannel := make(chan []logRecord)
+
+			var wait sync.WaitGroup
+
+			wait.Add(1)
+			go func() {
+				bf.decode(cmd.Context(), os.Stdin, recordChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.group(cmd.Context(), recordChannel, recordGroupsChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.write(cmd.Context(), recordGroupsChannel, os.Stdout)
+				wait.Done()
+			}()
+
+			wait.Wait()
+			return nil
+		},
+	}
+
+	cmd.Flags().IntVar(&groupSize, "group-size", 200,
+		"The maximum size of the log section to be sorted")
```

Comment:
"group" will be very extraneous to the end user.
How about "sorting window size" or something like that?
```suggestion
	cmd.Flags().IntVar(&groupSize, "sorting-window-size", 200,
		"The maximum size of the window where logs are collected for sorting")
```

Reply:
Probably better with "sorting-group-size"

Reply:
done

Reply:
@leonardoce I tested this picking the `sorting-window-size` as 0.
It actually works, but the effect is not to do grouping at all. Do we want to accept this?